### DISCUSSION TITLE
How to access path string (not location.pathname) in v6?

### DISCUSSION THREAD

**Discussion by arinthros** [remix-run/react-router#8378] [Q&A]
**How can I access the path string in v6?**

In v5 I could do `const { path, url } = useRouteMatch()` to get the un-substituted path, like `page/:pageId/edit`.
In v6 `useLocation` only returns the final path (formerly what `url` gave us `page/123/edit`.

I have poured over the docs and source code, and cannot seem to figure it out.

**Comment by liuhanqu** [comment]
https://reactrouter.com/docs/en/v6/api#useparams
It's what you want?

**Reply by arinthros** [reply]
No, sorry. I'm looking for the route string itself, not the param values. Something like `/page/:pageId/edit` which in v5 came from `useRouteMatch`.

**Comment by antonbehaeghe** [comment]
Ever managed to solve this? I'm having the same problem in v6.4
